tangelo/toolboxes/molecular_computation/integral_solver_pyscf.py

```diff
@@ -0,0 +1,193 @@
+
+        # Getting 2-body integrals in atomic and converting to molecular basis.
+        two_electron_integrals = self.ao2mo.kernel(pyscf_mol.intor("int2e"), mo_coeff)
+        two_electron_integrals = self.ao2mo.restore(1, two_electron_integrals, len(mo_coeff))
+
+        # PQRS convention in openfermion:
+        # h[p,q]=\int \phi_p(x)* (T + V_{ext}) \phi_q(x) dx
+        # h[p,q,r,s]=\int \phi_p(x)* \phi_q(y)* V_{elec-elec} \phi_r(y) \phi_s(x) dxdy
+        # The convention is not the same with PySCF integrals. So, a change is
+        # made before performing the truncation for frozen orbitals.
+        two_electron_integrals = two_electron_integrals.transpose(0, 2, 3, 1)
+
+        return core_constant, one_electron_integrals, two_electron_integrals
+
+    def compute_uhf_integrals(self, sqmol, mo_coeff):
+        """Compute 1-electron and 2-electron integrals
+        The return is formatted as
+        [numpy.ndarray]*2 numpy array h_{pq} for alpha and beta blocks
+        [numpy.ndarray]*3 numpy array storing h_{pqrs} for alpha-alpha, alpha-beta, beta-beta blocks
+
+        Args:
+            sqmol (SecondQuantizedMolecule): The SecondQuantizedMolecule object to calculated UHF integrals for.
+            mo_coeff (List[array]): The molecular orbital coefficients for both spins [alpha, beta]
+
+        Returns:
+            List[array], List[array]: One and two body integrals
+        """
+        # step 1 : find nao, nmo (atomic orbitals & molecular orbitals)
+
+        # molecular orbitals (alpha and beta will be the same)
+        # Lets take alpha blocks to find the shape and things
+
+        # molecular orbitals
+        nmo = mo_coeff[0].shape[1]
+        # atomic orbitals
+        nao = mo_coeff[0].shape[0]
+
+        # step 2 : obtain Hcore Hamiltonian in atomic orbitals basis
+        hcore = sqmol.mean_field.get_hcore()
+
+        # step 3 : obatin two-electron integral in atomic basis
+        eri = self.ao2mo.restore(8, sqmol.mean_field._eri, nao)
+
+        # step 4 : create the placeholder for the matrices
+        # one-electron matrix (alpha, beta)
+        hpq = []
+
+        # step 5 : do the mo transformation
+        # step the mo coeff alpha and beta
+        mo_a = mo_coeff[0]
+        mo_b = mo_coeff[1]
+
+        # mo transform the hcore
+        hpq.append(mo_a.T.dot(hcore).dot(mo_a))
+        hpq.append(mo_b.T.dot(hcore).dot(mo_b))
+
+        # mo transform the two-electron integrals
+        eri_a = self.ao2mo.incore.full(eri, mo_a)
+        eri_b = self.ao2mo.incore.full(eri, mo_b)
+        eri_ba = self.ao2mo.incore.general(eri, (mo_a, mo_a, mo_b, mo_b), compact=False)
+
+        # Change the format of integrals (full)
+        eri_a = self.ao2mo.restore(1, eri_a, nmo)
+        eri_b = self.ao2mo.restore(1, eri_b, nmo)
+        eri_ba = eri_ba.reshape(nmo, nmo, nmo, nmo)
+
+        # # convert this into the order OpenFemion like to receive
```

Comment:
`# Convert this (...)`

Let's not say "OpenFermion likes to receive" but instead use something like chemist / physicist notation if we have something like that